crates/ra_assists/src/assists/raw_string.rs

```diff
@@ -8,9 +9,27 @@ pub(crate) fn make_raw_string(mut ctx: AssistCtx<impl HirDatabase>) -> Option<As
     if literal.token().kind() != ra_syntax::SyntaxKind::STRING {
         return None;
     }
+    let token = literal.token();
+    let text = token.text().as_str();
+    let usual_string_range = find_usual_string_range(text)?;
     ctx.add_action(AssistId("make_raw_string"), "make raw string", |edit| {
         edit.target(literal.syntax().text_range());
-        edit.insert(literal.syntax().text_range().start(), "r");
+        let start_of_inside = usual_string_range.start().to_usize() + 1;
+        let end_of_inside = usual_string_range.end().to_usize();
+        let inside_str = &text[start_of_inside..end_of_inside];
+        let mut unescaped = String::with_capacity(inside_str.len());
+        let mut error = Ok(());
+        rustc_lexer::unescape::unescape_str(inside_str, &mut |_, unescaped_char| {
+            match unescaped_char {
+                Ok(c) => unescaped.push(c),
+                Err(_) => error = Err(()),
+            }
+        });
+        if error.is_err() {
+            eprintln!("Error unescaping string");
+        } else {
+            edit.replace(literal.syntax().text_range(), format!("r#\"{}\"#", unescaped));
```

Comment:
let's do a proper thing here and use `unescaped.chars().filter(|it| it == '#').count() + 1` hashes

Reply:
Ah, it should probably be smarter than that, as we shouldn't count the total number of hashes, but the longest `"#*` streak. Still shouldn't be hard to implement

Reply:
nice catch, I totally over sighted this use case :)